ztools/ztls/handshake_messages.go

```diff
@@ -545,7 +570,8 @@ func (m *serverHelloMsg) equal(i interface{}) bool {
 		m.ocspStapling == m1.ocspStapling &&
 		m.ticketSupported == m1.ticketSupported &&
 		m.secureRenegotiation == m1.secureRenegotiation &&
-		m.extendedMasterSecret == m1.extendedMasterSecret
+		m.extendedMasterSecret == m1.extendedMasterSecret &&
+		reflect.DeepEqual(m.unknownExtensions, m1.unknownExtensions)
```

Comment:
`unknownExtensions` is a 2D slice (every extension has its own slice of bytes), so unfortunately bytes.Equal cannot be used here.